pages/blog/extracting-all-mongodb-fields.mdx

```diff
@@ -0,0 +1,122 @@
+---
+title: "Extracting all MongoBD fields including the nested ones"
+date: "2021-04-06"
+author: "Paulo Ouriques"
+pullQuote: "A simple way to get all first-level fields and, potentially, all nested fields from a MongoDB collection."
+image: "bigquery-source/source-types.png"
+tags: [engineering]
+---
+
+If you’re a data analyst, data scientist, developer, or DB administrator you may have used, at some point, a non-relational database with flexible schemas. Well, I could list several advantages of a NoSQL solution over SQL-based databases. However, the main focus of this post is to discuss a particular downside of MongoDB and a possible solution to go through it.
+
+Recently, working on the Grouparoo source plugin for MongoDB, I came across a very simple requirement (simple, at least on SQL-based DBs with fixed schemas) which is to get all field names of a collection. However, this is not a simple query on MongoDB as it should be and we’re going to see what is needed to achieve this.
```

Comment:
Add a link to the plugin (npm or github, your choice)

Reply:
You could add a screenshot showing why you wanted the columns - perhaps from https://github.com/grouparoo/grouparoo/pull/1600